### ISSUE TITLE
Inline variable code action

### ISSUE THREAD

**Issue by lpil** [gleam-lang/gleam#3935]
An action that inlines a variable that is used once
```gleam
pub fn main() {
  let value = [1, 2, 3]
  list.map(value, run)
  #        ^^^^^
}
```

```gleam
pub fn main() {
  list.map([1, 2, 3], run)
}
```

**Comment by GearsDatapacks** [comment]
Inlining variables. It's not much of an optimisation by itself, but it allows for further optimisations (just like function inlining).
For example:
```gleam
let a = 1
a + 2
```
Becomes:
```gleam
1 + 2
```
Which the compiler should be able to infer as (although it doesn't currently):
```gleam
3
```

**Comment by giacomocavalieri** [comment]
Yeah but I think it's also very much a stylistic choice so having a code action to do it would be really nice

**Comment by GearsDatapacks** [comment]
Certainly

**Comment by lpil** [comment]
I'm not sure what the motivation is there. An optimising compiler may perform that action if it expects that structure to be able to apply later optimisations, but we don't implement and optimisations, and if we did it may not benefit from this.

**Comment by GearsDatapacks** [comment]
Yeah perhaps it isn't useful right now
